service: efi-serverless
frameworkVersion: '3'
configValidationMode: error
provider:
  name: aws
  architecture: ${param:architecture, 'x86_64'}
  runtime: python3.11
  stage: ${opt:stage, 'local'}
  region: ${opt:region, 'us-east-1'}
  timeout: 30
  environment:
    PYTHONPATH: '/var/task/lambdas:/opt/python/lib/site-packages' #/var/task/venv/lib/python3.11/site-packages:
    LOG_LEVEL: "DEBUG"
    LS_LOG: "trace"
    DEBUG: "1"
    ENVIRONMENT: LOCAL

layers: ${file(resources/layers.yml):layers}

plugins:
  - serverless-localstack
  - serverless-iam-roles-per-function
  - serverless-appsync-plugin
  - ./layers/layerInstallPlugin.js

custom:
  outputToEnv:
    map:
      VITE_USER_POOL_ID: UserPoolId
      VITE_GRAPHQL_API_URL: AppSyncUrl
      # VITE_REST_API_URL:
      VITE_REGION: Region
      VITE_HOSTED_UI: HostedUI
      VITE_USER_POOL_CLIENT_ID: UserPoolClientId
      VITE_LOCAL: Local
  localstack:
    host: http://127.0.0.1
    stages:
      - local
    lambda:
      mountCode: false
    autostart: false
  serverless-iam-roles-per-function:
    defaultInherit: true
  vpcConfig: ${file(resources/vpc.yml):vpcConfig}

appSync: ${file(resources/appSync.yml):appSync}

resources:
  - ${file(resources/secrets.yml)}
  - ${file(resources/db.yml)}
  - ${file(resources/iam.yml)}

functions:
  - ${file(resources/setup_functions.yml)}
  - ${file(resources/graphql_functions.yml)}

package:
  individually: true
  patterns:
    - '!**/**'
    - 'lambdas/__init__.py'
    - 'lambdas/utils/dt_utils.py'
    - 'lambdas/utils/str_utils.py'
    - 'lambdas/utils/validators.py'
    - 'lambdas/utils/common.py'
    - 'lambdas/services/db_manager.py'
    - 'lambdas/services/secrets_service.py'
